Formula/elan-init.rb

```diff
@@ -0,0 +1,51 @@
+class ElanInit < Formula
+  desc "Lean Theorem Prover installer and version manager"
+  homepage "https://leanprover-community.github.io/"
+  url "https://github.com/leanprover/elan/archive/v1.0.3.tar.gz"
+  sha256 "fb3ddd8915a0694ead0f3a51fdf8b0a5540f983f44eee0e757339244c522b8ee"
+  license "Apache-2.0"
+  head "https://github.com/leanprover/elan.git"
+
+  depends_on "rust" => :build
+
+  # elan-init will run on arm64 Macs, but will fetch Leans that are x86_64.
+  depends_on arch: :x86_64
+
+  depends_on "gmp"
+
+  def install
+    ENV["RELEASE_TARGET_NAME"] = "homebrew-build"
+
+    system "cargo", "install", "--features", "no-self-update", *std_cargo_args
+
+    %w[lean leanpkg leanchecker leanc leanmake elan].each do |link|
+      bin.install_symlink "elan-init" => link
+    end
+
+    bash_output = Utils.safe_popen_read("#{bin}/elan", "completions", "bash")
+    (bash_completion/"elan").write bash_output
+    zsh_output = Utils.safe_popen_read("#{bin}/elan", "completions", "zsh")
+    (zsh_completion/"_elan").write zsh_output
+    fish_output = Utils.safe_popen_read("#{bin}/elan", "completions", "fish")
+    (fish_completion/"elan.fish").write fish_output
+  end
+
+  test do
+    ENV["ELAN_HOME"] = testpath/".elan"
+
+    system bin/"elan-init", "-y"
+    (testpath/"hello.lean").write <<~EOS
+      def id' {α : Type} (x : α) : α := x
+
+      inductive tree (α : Type) : Type
+      | node : α → list tree → tree
+
+      example (a b : Prop) : a ∧ b -> b ∧ a :=
+      begin
+          intro h, cases h,
+          split, repeat { assumption }
+      end
+    EOS
+    system "#{bin}/lean", testpath/"hello.lean"
```

Comment:
```suggestion
    system bin/"lean", testpath/"hello.lean"
```
We don't need interpolation here. Is there output here we can check? Or does `lean` just return an exit code of zero if it compiles?

Reply:
This block is again copied exactly from the `lean.rb` formula :) but will make the change.

`lean` just exits without output if it compiles, yeah.